src/main/kotlin/app/revanced/patches/backdrops/misc/pro/patch/ProUnlockPatch.kt

```diff
@@ -0,0 +1,38 @@
+package app.revanced.patches.backdrops.misc.pro.patch
+
+import app.revanced.patcher.annotation.Description
+import app.revanced.patcher.annotation.Name
+import app.revanced.patcher.annotation.Version
+import app.revanced.patcher.data.BytecodeContext
+import app.revanced.patcher.extensions.MethodFingerprintExtensions.name
+import app.revanced.patcher.extensions.addInstructions
+import app.revanced.patcher.patch.BytecodePatch
+import app.revanced.patcher.patch.PatchResult
+import app.revanced.patcher.patch.PatchResultError
+import app.revanced.patcher.patch.PatchResultSuccess
+import app.revanced.patcher.patch.annotations.Patch
+import app.revanced.patches.backdrops.misc.pro.annotations.ProUnlockCompatibility
+import app.revanced.patches.backdrops.misc.pro.fingerprints.ProUnlockFingerprint
+
+@Patch
+@Name("pro-unlock")
+@Description("Unlocks pro-only functions.")
+@ProUnlockCompatibility
+@Version("0.0.1")
+class ProUnlockPatch : BytecodePatch(
+    listOf(ProUnlockFingerprint)
+) {
+    override fun execute(context: BytecodeContext): PatchResult {
+        val result = ProUnlockFingerprint.result ?: return PatchResultError("${ProUnlockFingerprint.name} not found")
+
+        result.mutableMethod.addInstructions(
+            result.scanResult.patternScanResult!!.endIndex,
+            """
+                const/4 v4, 0x6
+                const/4 v2, 0x1
```

Comment:
The register 4 and 2 might change, are you confident they wont?

Reply:
No, I don't think they'll change. the app haven't had an update for 2 years. But who knows

Reply:
Also, what can I do in this kind of situation? I started learning smali Bytecode yesterday, so I don't know very much of this

Reply:
You'd dynamically get the registers based on the current context, such as existing instructions which use them.